Support _source field in mappings editor

cjcenizal opened the pull request:
Fixes https://github.com/elastic/kibana/issues/53035 and fixes https://github.com/elastic/kibana/issues/52684

CC @sebelga and @alisonelizabeth I'd like your input on some of the refactoring I've done here.

Beyond the addition of the source field form I've made some small architectural changes which have made the code easier for me to follow and I'd like to make sure they're also helpful/neutral to others:

* Moved `CONFIGURATION_FIELDS` into the configuration schema file.
* Migrated from the documentation config object to defining the link in the documentations service. There are fewer inputs for me to consider (the documentation service and component that renders the doc link) when trying to figure out the output (a rendered doc link).
* Used named exports instead of `*` to make easier to trace the parts of module's interface back to their source.

## Tabs

### Fields

![image](https://user-images.githubusercontent.com/1238659/70864213-d9a4ae00-1f04-11ea-8386-5a2786429688.png)

### Advanced

![image](https://user-images.githubusercontent.com/1238659/70864219-e6290680-1f04-11ea-9c5b-0f228231959a.png)

## Dynamic mapping

### Enabled

![image](https://user-images.githubusercontent.com/1238659/70864034-3bfcaf00-1f03-11ea-8fdb-4288f0149a0e.png)

### Disabled

![image](https://user-images.githubusercontent.com/1238659/70864040-4028cc80-1f03-11ea-9a79-1f4e5b0a25f8.png)

## Source field

### Enabled

![image](https://user-images.githubusercontent.com/1238659/70864019-14a5e200-1f03-11ea-8d43-ea519115cac0.png)

### Disabled

![image](https://user-images.githubusercontent.com/1238659/70864023-21c2d100-1f03-11ea-851c-79fb60a0cfbc.png)

cjcenizal committed:
Extract dynamic mapping section into its own component. Migrate doc link from schema to documentation service.

cjcenizal committed:
Add sourceFieldSchema and SourceFieldForm for enabling and disabling _source.
- Callout for warning when _source is disabled.
- Combo boxes for including and excluding fields.

elasticmachine commented:
Pinging @elastic/es-ui (Team:Elasticsearch UI)

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15573/)
* Commit: aa427c7548f61136244293cd46ff49ee500cba5f

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15573/","number":"15573","commit":"aa427c7548f61136244293cd46ff49ee500cba5f"}]}
PIPELINE-->

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15592/)
* Commit: 23901c230788138299f550303e14af2e9042bfaa

### History
* :broken_heart: [Build #15573](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15573/) failed aa427c7548f61136244293cd46ff49ee500cba5f

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15573/","number":"15573","commit":"aa427c7548f61136244293cd46ff49ee500cba5f"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15592/","number":"15592","commit":"23901c230788138299f550303e14af2e9042bfaa"}]}
PIPELINE-->

cjcenizal committed:
Split content into Fields and Advanced options tabs.

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15602/)
* Commit: f70627ccae58a642ea9919369243a197b54dee1b

### History
* :broken_heart: [Build #15592](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15592/) failed 23901c230788138299f550303e14af2e9042bfaa
* :broken_heart: [Build #15573](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15573/) failed aa427c7548f61136244293cd46ff49ee500cba5f

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15573/","number":"15573","commit":"aa427c7548f61136244293cd46ff49ee500cba5f"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15592/","number":"15592","commit":"23901c230788138299f550303e14af2e9042bfaa"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15602/","number":"15602","commit":"f70627ccae58a642ea9919369243a197b54dee1b"}]}
PIPELINE-->

cjcenizal committed:
Remove unnecessary padding from edit_field form.

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15611/)
* Commit: f585909da013c57539049bf9aa149b45a44503e3

### History
* :broken_heart: [Build #15602](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15602/) failed f70627ccae58a642ea9919369243a197b54dee1b
* :broken_heart: [Build #15592](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15592/) failed 23901c230788138299f550303e14af2e9042bfaa
* :broken_heart: [Build #15573](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15573/) failed aa427c7548f61136244293cd46ff49ee500cba5f

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15573/","number":"15573","commit":"aa427c7548f61136244293cd46ff49ee500cba5f"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15592/","number":"15592","commit":"23901c230788138299f550303e14af2e9042bfaa"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15602/","number":"15602","commit":"f70627ccae58a642ea9919369243a197b54dee1b"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15611/","number":"15611","commit":"f585909da013c57539049bf9aa149b45a44503e3"}]}
PIPELINE-->

cjcenizal committed:
Update form hook lib components to accept helpText config as either string or function.

cjcenizal committed:
Add toggle for disabling dynamic mapping. Refine dynamic mapping UI and help text.

cjcenizal committed:
Refine copy:
- Consistently prefix _source field with underscore.
- Rename Fields tab to Mapped fields.

cjcenizal commented:
Note that there is a known bug where clicking "Edit" on a field and then clicking the "Advanced options" tab results in clicks not having any effect in the UI.

![image](https://user-images.githubusercontent.com/1238659/70864278-80894a00-1f05-11ea-998d-c8d9d5011399.png)

It might make sense to address this bug as part of https://github.com/elastic/kibana/issues/53012.

cjcenizal commented:
Just spotted another bug. @sebelga I wonder if this could be a problem in the form hook lib? It looks like the states of two components are interfering with one another, but I don't see any cause for it in the code. Is the state of a component tracked by its order in the form, or by its key?

![dynamic_form_bug](https://user-images.githubusercontent.com/1238659/70864633-c4318300-1f08-11ea-8cea-da787e9d02dc.gif)

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15635/)
* Commit: 3d6420d381665ed0972fec4fb1c6cc022b97eb33

### History
* :broken_heart: [Build #15611](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15611/) failed f585909da013c57539049bf9aa149b45a44503e3
* :broken_heart: [Build #15602](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15602/) failed f70627ccae58a642ea9919369243a197b54dee1b
* :broken_heart: [Build #15592](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15592/) failed 23901c230788138299f550303e14af2e9042bfaa
* :broken_heart: [Build #15573](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15573/) failed aa427c7548f61136244293cd46ff49ee500cba5f

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15573/","number":"15573","commit":"aa427c7548f61136244293cd46ff49ee500cba5f"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15592/","number":"15592","commit":"23901c230788138299f550303e14af2e9042bfaa"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15602/","number":"15602","commit":"f70627ccae58a642ea9919369243a197b54dee1b"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15611/","number":"15611","commit":"f585909da013c57539049bf9aa149b45a44503e3"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15635/","number":"15635","commit":"3d6420d381665ed0972fec4fb1c6cc022b97eb33"}]}
PIPELINE-->

alisonelizabeth commented on x-pack/legacy/plugins/index_management/public/app/components/mappings_editor/components/configuration_form/source_field_form/source_field_schema.ts:
```diff
@@ -0,0 +1,56 @@
+    },
+  },
+  includes: {
+    label: i18n.translate('xpack.idxMgmt.mappingsEditor.excludeSourceFieldsLabel', {
```
should this be `includeSourceFieldsLabel`?

alisonelizabeth commented on x-pack/legacy/plugins/index_management/public/app/components/mappings_editor/components/configuration_form/source_field_form/source_field_schema.ts:
```diff
@@ -0,0 +1,56 @@
+    ...fieldPathComboBoxConfig,
+  },
+  excludes: {
+    label: i18n.translate('xpack.idxMgmt.mappingsEditor.includeSourceFieldsLabel', {
```
should this be `excludeSourceFieldsLabel`?

alisonelizabeth commented on x-pack/legacy/plugins/index_management/public/app/components/mappings_editor/components/document_fields/document_fields_header.tsx:
```diff
@@ -15,7 +15,7 @@ export const DocumentFieldsHeader = () => {
         <h2>
           {i18n.translate('xpack.idxMgmt.mappingsEditor.documentFieldsTitle', {
-            defaultMessage: 'Document fields',
+            defaultMessage: 'Mapped fields',
```
I noticed it's a little repetitive now with the tab and title. Thoughts?

<img width="194" alt="Screen Shot 2019-12-16 at 12 01 17 PM" src="https://user-images.githubusercontent.com/5226211/70927401-e1477e00-1ffc-11ea-9ba5-dae2695d24ef.png">

alisonelizabeth commented on x-pack/legacy/plugins/index_management/public/app/services/documentation.ts:
```diff
@@ -38,7 +37,7 @@ class DocumentationService {
   }
 
   public getTypeDocLink = (type: DataType | ConfigType, uri = 'main'): string | undefined => {
```
I think this can be cleaned up more if it is no longer going to be used for the config doc links.

```
  public getTypeDocLink = (type: DataType, uri = 'main'): string | undefined => {
    const typeDefinition = TYPE_DEFINITION[type];

    if (!typeDefinition || !typeDefinition.documentation || !typeDefinition.documentation[uri]) {
      return undefined;
    }
    return `${this.esDocsBase}${typeDefinition.documentation[uri]}`;
  };
```

alisonelizabeth reviewed (commented):
@cjcenizal nice job! This looks great. I came across the bugs you commented on. I also noticed unexpected behavior related to the defaults. I think this relates to https://github.com/elastic/kibana/issues/52708.

If I don't select the "Advanced" tab, the request looks like:

```
{
  "index_patterns": [
    "adsf"
  ],
  "mappings": {
    "dynamic": false,
    "_source": {},
    "properties": {}
  }
}
```

Whereas, if I do, the request looks like:
```
{
  "index_patterns": [
    "adsf"
  ],
  "mappings": {
    "dynamic": true,
    "numeric_detection": false,
    "date_detection": true,
    "dynamic_date_formats": [
      "strict_date_optional_time",
      "yyyy/MM/dd HH:mm:ss Z||yyyy/MM/dd Z"
    ],
    "_source": {
      "includes": [],
      "excludes": []
    },
    "properties": {}
  }
}
```

loixlab committed:
Code review changes recommendation

loixlab reviewed (changes requested):
Awesome job @cj! I have a few comments about the design decisions:

* First of all, I really like the tabs! :blush: It makes it all clearer, thanks for doing this.

* When I first got on the page, editing one template that only had 1 field, it seemed to me that the mappings editor lacked contrast and my single field was lost in the "white" around 2 lines.

<img width="890" alt="Screen Shot 2019-12-17 at 10 55 18" src="https://user-images.githubusercontent.com/2854616/70995218-066be780-20f6-11ea-8fc4-ca76256fbb63.png">


I added a background-color to the main `<ul class="mappingsEditor__fieldsList">` with the same color as the multi-steps and it stands out already much more to me.
As a second impression, I wondered why adding (in the copy) "Mapped" to "fields" and "advanced" to "options". I don't think we should scare too much the users too early. The real "_advanced_" settings the "_source" section (I'll come to it below). So what do you think of this instead?

<img width="893" alt="Screen Shot 2019-12-17 at 11 03 38" src="https://user-images.githubusercontent.com/2854616/70995237-0ff54f80-20f6-11ea-949b-ce31d11f1217.png">

* For the `_source` section, the doc indicates: "An expert-only feature is the ability to prune the contents of...". So not only advanced.. expert! :smile:

What do you think about:

  * moving the "Enable _source field" toggle to the right column
  * Have another toggle below (only visible if _source enabled) saying "Add all fields" (enabled by default)
  * If the user disables the "Add all fields", then a callout appears below warning of the downside (_Removing fields from the `_source` has similar downsides to disabling `_source`, especially the fact that you cannot reindex documents from one Elasticsearch index to another._)

* When disabling the dynamic mapping, I would prefer to be consistent and have a toggle instead of a checkbox for the "Throw an exception..." WDYT?

<img width="434" alt="Screen Shot 2019-12-17 at 11 05 59" src="https://user-images.githubusercontent.com/2854616/70995310-429f4800-20f6-11ea-8e25-b26396833910.png">


* As a first time user of the `_source` feature, I don't know why the UI says "There aren't any options available" when trying to include fields, as I _do_ have some fields defined.

<img width="545" alt="Screen Shot 2019-12-17 at 11 32 32" src="https://user-images.githubusercontent.com/2854616/70995342-55b21800-20f6-11ea-81e9-334abf1512dc.png">

<img width="436" alt="Screen Shot 2019-12-17 at 11 32 43" src="https://user-images.githubusercontent.com/2854616/70995359-5b0f6280-20f6-11ea-8442-cb26387349f5.png">


[Edit]: After looking at the code I understand what we do, but as a user it was confusing. I would expect to have the list of fields to chose from in the dropdown.

* About having a dedicagted `<SourceFieldForm />` component: if we look at the page, the dynamic mappings and the source field logically belong to the same form. It is strange to have created separate form and it adds complexity IMO. The great benefit of having a flat structure and nested field paths (with dot notation) in the form lib is not to have to worry about the outputted object structure.
So I would have one `<ConfigurationForm />`  with 2 form rows and the fields on the right. Then `enabled` becomes `_source.enabled`.
This means less moving parts, less reducer and not having to combine the form data ...

I will push a commit to show you how I would see it. Let me know your thoughts.

loixlab committed:
Add validator for space in field name

sebelga commented on x-pack/legacy/plugins/index_management/public/app/components/mappings_editor/components/configuration_form/configuration_form.tsx:
```diff
@@ -5,35 +5,41 @@
 }
 
-const UseField = getUseField({ component: Field });
+const formSerializer: SerializerFunc<MappingsConfiguration> = formData => {
```
Here is where we build our final form data.

sebelga commented on x-pack/legacy/plugins/index_management/public/app/components/mappings_editor/components/configuration_form/configuration_form_schema.tsx:
```diff
@@ -0,0 +1,131 @@
+      },
+    ],
+  },
+  _source: {
```
We can use object in the schema for nested props.

sebelga commented on x-pack/legacy/plugins/index_management/public/app/components/mappings_editor/components/configuration_form/source_field.tsx:
```diff
@@ -0,0 +1,159 @@
+} from '../../shared_imports';
+import { ComboBoxOption } from '../../types';
+
+const i18nTexts = {
```
I like to have the i18n text out of the component as it clutters the JSX and it's harder to follow what is important. Here I can simply fold this `i18nTexts` in the IDE and focus on the component logic.

sebelga commented on x-pack/legacy/plugins/index_management/public/app/components/mappings_editor/components/configuration_form/source_field.tsx:
```diff
@@ -0,0 +1,159 @@
+  const { setValue } = field;
+
+  return (
+    <ComboBoxField
```
The `ComboBoxField` takes care of almost everything. We just need to override below what is unique in this context.

sebelga commented on x-pack/legacy/plugins/index_management/public/app/components/mappings_editor/mappings_editor.tsx:
```diff
@@ -25,42 +26,69 @@ interface Props {
+      () => ({
+        configurationDefaultValue: {
+          ...(pick(defaultValue, CONFIGURATION_FIELDS) as Types['MappingsConfiguration']),
+          _source: defaultValue._source || {},
```
The `_source` is added to the configuration object  default value.

sebelga commented on x-pack/legacy/plugins/index_management/public/app/components/mappings_editor/mappings_state.tsx:
```diff
@@ -111,13 +112,15 @@ export const MappingsState = React.memo(({ children, onUpdate, defaultValue }: P
+            : deNormalize(nextState.fields);
+
         return {
           ...nextState.configuration.data.format(),
```
This stays as it was.

sebelga commented on x-pack/legacy/plugins/index_management/public/app/services/documentation.ts:
```diff
@@ -56,9 +55,25 @@ class DocumentationService {
     return `${this.esDocsBase}/mapping-types.html`;
   }
 
+  public getDynamicMappingLink() {
```
I am not quite sure about all those methods but it deserves that we talk about it over zoom to explain my reasoning. Let's leave them for now 😊

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/16058/)
* Commit: 7a7fb71d3613b04f5fc69fc0c38bffc408b29b00

### History
* :broken_heart: [Build #15635](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15635/) failed 3d6420d381665ed0972fec4fb1c6cc022b97eb33
* :broken_heart: [Build #15611](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15611/) failed f585909da013c57539049bf9aa149b45a44503e3
* :broken_heart: [Build #15602](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15602/) failed f70627ccae58a642ea9919369243a197b54dee1b
* :broken_heart: [Build #15592](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15592/) failed 23901c230788138299f550303e14af2e9042bfaa
* :broken_heart: [Build #15573](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15573/) failed aa427c7548f61136244293cd46ff49ee500cba5f

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15573/","number":"15573","commit":"aa427c7548f61136244293cd46ff49ee500cba5f"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15592/","number":"15592","commit":"23901c230788138299f550303e14af2e9042bfaa"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15602/","number":"15602","commit":"f70627ccae58a642ea9919369243a197b54dee1b"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15611/","number":"15611","commit":"f585909da013c57539049bf9aa149b45a44503e3"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15635/","number":"15635","commit":"3d6420d381665ed0972fec4fb1c6cc022b97eb33"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/16058/","number":"16058","commit":"7a7fb71d3613b04f5fc69fc0c38bffc408b29b00"}]}
PIPELINE-->

patrykkopycinski committed:
[SIEM] Cleanup _g param from url (#53316)

patrykkopycinski committed:
[SIEM] Fix eslint errors from jsx-no-bind #1 (#52856)

simianhacker committed:
[Metrics UI] Display data label on inventory based on data interval (#53054)

FrankHassanabad committed:
[SIEM][Detection Engine] Adds prepackaging rules capability

## Summary

* Adds pre-packaging rules capability
* Adds pre-packaged rules 
* Adds conversion script for saved searches to pre-package then edit workflow
* Adds an auto-incrementing numbered versioning
* Adds `immutability` as a filterable internal tag
 
<img width="1833" alt="Screen Shot 2019-12-14 at 12 21 07 AM" src="https://user-images.githubusercontent.com/1151048/70845308-5010b580-1e0a-11ea-833e-8d5656b49a98.png">

<img width="1762" alt="Screen Shot 2019-12-14 at 12 23 40 AM" src="https://user-images.githubusercontent.com/1151048/70845311-569f2d00-1e0a-11ea-866c-5aa578377b15.png">

Usage/Testing:


```sh
PUT /api/detection_engine/rules/prepackaged
```

script:
```sh
./scripts/add_prepackaged_rules.sh
```

When you create a new rule now (regardless of if it is immutable or not), you automatically get a version number returned back starting at the number 1. If you update the rule then it will automatically update the version number by incrementing it by 1 for convenience.

However if you create a rule _and_ add in the version number during the creation of the rule then it will take that version number. If you update a rule you can optionally send in the version which will also update it there. 

### Checklist

Use ~~strikethroughs~~ to remove checklist items you don't feel are applicable to this PR.

~~- [ ] This was checked for cross-browser compatibility, [including a check against IE11](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility)~~

~~- [ ] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/master/packages/kbn-i18n/README.md)~~

~~- [ ] [Documentation](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#writing-documentation) was added for features that require explanation or tutorials~~

- [x] [Unit or functional tests](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility) were updated or added to match the most common scenarios

~~- [ ] This was checked for [keyboard-only and screenreader accessibility](https://developer.mozilla.org/en-US/docs/Learn/Tools_and_testing/Cross_browser_testing/Accessibility#Accessibility_testing_checklist)~~

### For maintainers

~~- [ ] This was checked for breaking API changes and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)~~

- [x] This includes a feature addition or change that requires a release note and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)

simianhacker committed:
[Metrics UI] Fix AWS SQS toolbar types (#53051)

Commit:
[yarn] upgrade caniuse-lite database (#53321)

cjcenizal committed:
Address Alison's PR feedback.

Commit:
[Maps] Update license check & requests to use NP services (#52641)

* Add back in np licensing changes previously pulled out of a separate pr

* Watch license service and update local kibana service const obj. on change. Import new license id in meta

* Update flow for initializing indexPatternServices while in this file

* Clean up setup method. Add conditional subscribe of licensing

* Review feedback. Use license getter instead of obj

* Revert "Update flow for initializing indexPatternServices while in this file"

This reverts commit 022f7ddb87b565d2962c1ef28a6e0620afbc570b.

* Update meta test

FrankHassanabad committed:
[SIEM][Detection Engine] Adds timeline_id string to rules (#53343)

## Summary

* Adds the timeline_id string to rules
* Updates unit tests and fixes a few issues in the unit tests to make types more strict
* Updates signal mapping to have it

### Checklist

Use ~~strikethroughs~~ to remove checklist items you don't feel are applicable to this PR.

~~- [ ] This was checked for cross-browser compatibility, [including a check against IE11](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility)~~

~~- [ ] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/master/packages/kbn-i18n/README.md)~~

~~- [ ] [Documentation](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#writing-documentation) was added for features that require explanation or tutorials~~

- [x] [Unit or functional tests](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility) were updated or added to match the most common scenarios

~~- [ ] This was checked for [keyboard-only and screenreader accessibility](https://developer.mozilla.org/en-US/docs/Learn/Tools_and_testing/Cross_browser_testing/Accessibility#Accessibility_testing_checklist)~~

### For maintainers

~~- [ ] This was checked for breaking API changes and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)~~

- [x] This includes a feature addition or change that requires a release note and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/16315/)
* Commit: 6f1178c832d5372a95f053843cc97b2d42d54fb4

### History
* :broken_heart: [Build #16058](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/16058/) failed 7a7fb71d3613b04f5fc69fc0c38bffc408b29b00
* :broken_heart: [Build #15635](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15635/) failed 3d6420d381665ed0972fec4fb1c6cc022b97eb33
* :broken_heart: [Build #15611](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15611/) failed f585909da013c57539049bf9aa149b45a44503e3
* :broken_heart: [Build #15602](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15602/) failed f70627ccae58a642ea9919369243a197b54dee1b
* :broken_heart: [Build #15592](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15592/) failed 23901c230788138299f550303e14af2e9042bfaa

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15592/","number":"15592","commit":"23901c230788138299f550303e14af2e9042bfaa"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15602/","number":"15602","commit":"f70627ccae58a642ea9919369243a197b54dee1b"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15611/","number":"15611","commit":"f585909da013c57539049bf9aa149b45a44503e3"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15635/","number":"15635","commit":"3d6420d381665ed0972fec4fb1c6cc022b97eb33"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/16058/","number":"16058","commit":"7a7fb71d3613b04f5fc69fc0c38bffc408b29b00"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/16315/","number":"16315","commit":"6f1178c832d5372a95f053843cc97b2d42d54fb4"}]}
PIPELINE-->

cjcenizal committed:
Prevent sending request for rollup indices if we're on an anonymous path, e.g. /logout. (#53233)

mistic committed:
[build] Add UBI7 docker target (#52735)

* feat(NA): add ubi based docker build images step

* chore(NA): remove ubiImageFlavor from appending to oss dir

cjcenizal committed:
Add support for xpack.license_management.ui.enabled setting. (#53209)

alisonelizabeth commented:
If this PR is going to take more time, would it make sense to create a separate PR for the form lib changes?

cjcenizal commented:
> If this PR is going to take more time, would it make sense to create a separate PR for the form lib changes?

This PR has a bad bug in it, in which the "Add field" form is no longer showing up and the "Add field" button is a no-op. I'm working on it now but I'm having a hard time pinning down where this bug is coming from. In the future I think we should strive to keep our PRs granular and short-lived, and introducing form lib changes in separate PRs sounds like a good way to do this.

Here's what the bug looks like:

![image](https://user-images.githubusercontent.com/1238659/71117763-a031a400-218b-11ea-915e-1a7fcda937ae.png)

cjcenizal committed:
Revert "Address Alison's PR feedback."

This reverts commit 6f1178c832d5372a95f053843cc97b2d42d54fb4.

cjcenizal committed:
Revert "Add validator for space in field name"

This reverts commit 7a7fb71d3613b04f5fc69fc0c38bffc408b29b00.

cjcenizal committed:
Revert "Code review changes recommendation"

This reverts commit b61a1baa6f3beff15b26e167c677e189472204c7.

cjcenizal committed:
Address Alison's feedback:
- Remove title from document fields header.
- Fix typos in i18n keys and text.
- Simplify getTypeDocLink function.

cjcenizal commented:
Sorry for the noise everyone! I messed up a rebase of the target branch. Closing this in favor of https://github.com/elastic/kibana/pull/53060

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/16545/)
* Commit: 58e4a60ad76727ab3bb2a3b3c0a887fa480a9252

### History
* :broken_heart: [Build #16315](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/16315/) failed 6f1178c832d5372a95f053843cc97b2d42d54fb4
* :broken_heart: [Build #16058](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/16058/) failed 7a7fb71d3613b04f5fc69fc0c38bffc408b29b00
* :broken_heart: [Build #15635](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15635/) failed 3d6420d381665ed0972fec4fb1c6cc022b97eb33
* :broken_heart: [Build #15611](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15611/) failed f585909da013c57539049bf9aa149b45a44503e3
* :broken_heart: [Build #15602](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15602/) failed f70627ccae58a642ea9919369243a197b54dee1b

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15602/","number":"15602","commit":"f70627ccae58a642ea9919369243a197b54dee1b"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15611/","number":"15611","commit":"f585909da013c57539049bf9aa149b45a44503e3"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15635/","number":"15635","commit":"3d6420d381665ed0972fec4fb1c6cc022b97eb33"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/16058/","number":"16058","commit":"7a7fb71d3613b04f5fc69fc0c38bffc408b29b00"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/16315/","number":"16315","commit":"6f1178c832d5372a95f053843cc97b2d42d54fb4"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/16545/","number":"16545","commit":"58e4a60ad76727ab3bb2a3b3c0a887fa480a9252"}]}
PIPELINE-->
